File: 0344-reverse-string/0344-reverse-string.ts
/**
 Do not return anything, modify s in-place instead.
 */
function reverseString(s: string[]): void {
    

    function helper(left: number, right:number): void{

        // base case
        if(left >= right) return

        // general case
        [s[left], s[right]] = [s[right], s[left]]

        // recurrance relation
        helper(left+1, right-1)


    }

    helper(0, s.length -1)
};

File: 0344-reverse-string/0344-reverse-string.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        # solving using recursion
        
        # create helper function
        def helper(left: int, right: int)-> None:
            # base case
            if left >= right:
                return
            
            # other cases
            s[left], s[right] = s[right], s[left]
            
            helper(left+1, right-1)
        
        helper(0, len(s)-1)
